src/main/java/view/ListView.java

```
@@ -254,31 +252,36 @@ void removeSelectedMovie() {
      * Moves selected movie to a different list.
      */
     private void moveSelectedMovie() {
-        final int selectedRow = mediaListTable.getSelectedRow();
-        if (selectedRow == -1) {
-            System.err.println("No movie selected. Cannot move.");
+        if (mediaListTable == null || listViewModel == null || mediaCollectionSelector == null
+                || moveController == null) {
+            System.err.println("Initialization error. Required components are missing.");
```

you could consider throwing an exception or returning early to fail fast